### ISSUE TITLE
Version info for downloaded releases not generated

### ISSUE THREAD

**Issue by holiman** [Proxmark/proxmark3#12]
Now that we have github versioning, all versions will be automatically available for download as zip. These zip-files contain the source for manual building, but this means that users will start compiling without git-integration. 
Thus, the builds will not contain version info.. 

How to solve? 
- Some kind of hooks on version-tag-commits to automatically place version number in a file? 

We probably need version info in some kind of static resource, question is how we generate that resource and when.

**Comment by iceman1001** [comment]
time to close this issue?

**Comment by ikarus23** [comment]
It's not resolved. If you download the source code to a release and try to compile it without having `git` installed, the `tools/mkversion.pl` script is most likely to fail (see https://github.com/Proxmark/proxmark3/blob/master/tools/mkversion.pl#L14).

**Comment by digitalentropy** [comment]
Just a thought...

The zip file that you can download from the main github page saves all of the files with the same timestamp as the last commit.  Why not just use that creation/modification date?

**Comment by iceman1001** [comment]
two usecases,  and existing solution solves one usecase.

1. ppl uses Git to pull the lastest source.
2. ppl download the zipfile

In 1, it doesnt update the timestamp on all files and in 2, can't get git commit info.

**Comment by digitalentropy** [comment]
Correct.  It wouldn't be consistent, but it would be perhaps better than nothing.  For those who use git, they get the hash.  For those that download the zipfile, their designator is the date code instead.

I mean, the purpose is principally to identify the version of firmware running on the device, correct?

**Comment by iceman1001** [comment]
could update the mkversion.pl,  to default down to timestamp of a file, when git command returns nothing.
The file timestamp only identify when it was downloaded,  not which version.

**Comment by digitalentropy** [comment]
It identifies the last commit.  I just tested it by downloading the zip and unzipping it.  The date stamp on all of the files within the zip matches the date of the last commit, 4/28.

**Comment by iceman1001** [comment]
Then that would work as an fallback option,  the datestamp,  someone just needs to update perlscript mkversion.pl

**Comment by iceman1001** [comment]
@digitalentropy  I've pushed a fix to icemanfork for this one.  Would you mind testing it out?

**Comment by iceman1001** [comment]
well..  you could take the fix from iceman fork,   and this would be solved....   *just sayin'*

**Comment by yaxax** [comment]
I can't install proxmark3 on Kali I always get no space left on device.And I have nothing install in the Kali. I only try to install the proxmark3 can somebody help me out thanks

**Comment by pwpiwi** [comment]
This question is not at all related to the source code on this github repository. Please ask this question in the forum http://www.proxmark.org/forum/index.php
